### PR TITLE
Check that rmw_init() fails if no enclave is given.

### PR DESCRIPTION
**Pull request by hidmic**
Connected to https://github.com/ros2/rmw/pull/247.

### CODE REVIEW

**Comment by hidmic** [comment]
Running CI up to `test_rmw_implementation` and all RMW implementations:

* Linux [Build Status](http://ci.ros2.org/job/ci_linux/11382/)

File: test_rmw_implementation/test/test_init_shutdown.cpp [hunk 1]
```diff
@@ -34,6 +35,8 @@ class CLASSNAME (TestInitShutdown, RMW_IMPLEMENTATION) : public ::testing::Test
     options = rmw_get_zero_initialized_init_options();
     rmw_ret_t ret = rmw_init_options_init(&options, rcutils_get_default_allocator());
     ASSERT_EQ(RMW_RET_OK, ret) << rcutils_get_error_string().str;
+    options.enclave = rcutils_strdup("/", rcutils_get_default_allocator());
```
Comment (Blast545) [inline test_rmw_implementation/test/test_init_shutdown.cpp:38]: Shouldn't this allocated string be deallocated?

Comment (hidmic) [reply test_rmw_implementation/test/test_init_shutdown.cpp:38]: It'll be taken care of by `rmw_init_options_fini()`.

Comment (Blast545) [reply test_rmw_implementation/test/test_init_shutdown.cpp:38]: Just checked, you are right!

**Comment by jacobperron** [comment]
@ros-pull-request-builder retest this please

**Comment by dirk-thomas** [comment]
@ros-pull-request-builder retest this please

**Pull request merged**